#39: Add Popover from labonline (semantic free!) (closes #39)

FrancescoCioria opened the pull request:
Issue #39

francescogior commented on src/popover/Popover.js:
```diff
@@ -1,99 +1,308 @@
+        dismissOnScroll: true,
+        className: ''
+      },
+      ...this.props.popover
```
`{ ...{ prop1: value1, prop2: value2 }, ...rest }` should be the same of `{ prop1: value1, prop2: value2 , ...rest }`

FrancescoCioria replied:
:+1:

francescogior commented on src/popover/Popover.js:
```diff
@@ -1,99 +1,308 @@
+    this.saveValuesFromNodeTree();
+  },
+
+  saveValuesFromNodeTree(cb) {
```
Here and for all the methods shouldn't be safer to use fat arrows?

FrancescoCioria replied:
Nope as in react components we're not using ES6 classes

francescogior commented:
:+1:

FrancescoCioria commented:
@giogonzo maybe you want to give it a look too as we'll be using it in LOL soon

giogonzo commented:
looks good to me

- can you summarize additions with respect to LOL popover? is it just handling both absolute and relative to DOM?
- looks safe for iso rendering, but could you try it once? It should be as easy as:
```js
document.getElementById('container').innerHTML = React.renderToString(template);
// instead of:
// React.render(template, document.getElementById('container'));
```
(actually this could be a nice test for each component, maybe rendered side by side in examples?)

FrancescoCioria commented:
I also changed the API: I grouped all the props related to popover inside an object
see here: https://github.com/buildo/react-components/commit/861b6cf69c9324982b0b43ec91f5b7ee4d50185a

I added a global isomorphic test as suggested. It works fine

giogonzo commented:
nice
